controllers/apps/operations/horizontal_scaling.go

```diff
@@ -66,6 +66,16 @@ func (hs horizontalScalingOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli
 		}
 		r := horizontalScaling.Replicas
 		opsRes.Cluster.Spec.ComponentSpecs[index].Replicas = r
+		if horizontalScaling.Instances != nil {
+			opsRes.Cluster.Spec.ComponentSpecs[index].Instances = horizontalScaling.Instances
+		} else {
+			opsRes.Cluster.Spec.ComponentSpecs[index].Instances = nil
+		}
+		if horizontalScaling.Nodes != nil {
```

Comment:
My understanding is that nodes are a resource pool, and pods will select nodes in order.

Now there is another scenario where a cluster of 3 nodes and 3 replicates.
If specify 2 nodes and scale out to 5 replicas, the new pod will be scheduled to the new node.
But if a restart occurs at this time, according to the algorithm for allocating pods, it will result in all 5 pods being allocated to these 2 nodes.

 Does the above scenario meet expectations?

Reply:
get it